### PR TITLE
DOCS-#3300: Add section about OmniSci engine with initial info

### LINKED ISSUE #3300: Add documentation for OmniSci engine
We need to add information about OmniSci engine (how to install, enable in code, supported operations etc) to modin.readthedocs.io

### PR DESCRIPTION
**Pull request by Garra1980**
Signed-off-by: izamyati <igor.zamyatin@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3300  
- [ ] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by YarShev** [comment]
@Garra1980 , please, create an issue that should be resolved as part of this PR.

File: docs/UsingOmnisci/index.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
+OmniSci
```
Comment (YarShev) [inline docs/UsingOmnisci/index.rst:1]: Until we don't exclude Ray dependency out of the OmniSci engine, we should probably use the former in the names of the backend (`OmnisciOnRay`).

File: docs/UsingOmnisci/index.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+=======
+
+This section describes usage related documents for the OmniSciDB-based engine of Modin.
```
Comment (YarShev) [inline docs/UsingOmnisci/index.rst:4]:
```suggestion
This section describes usage of related documents for the OmniSciDB-based engine of Modin.
```

File: docs/UsingOmnisci/index.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+This engine uses analytical database OmniSciDB_ to obtain high single-node scalability for
+specific set of datframe operations.
```
Comment (YarShev) [inline docs/UsingOmnisci/index.rst:7]:
```suggestion
specific set of dataframe operations.
```

File: docs/index.rst [hunk 4]
```diff
@@ -170,6 +170,7 @@ and lots of other useful information.**
 
    UsingPandasonRay/index
    UsingPandasonDask/index
+   UsingOmnisci/index
```
Comment (YarShev) [inline docs/index.rst:173]: We should also add something to this section.
https://modin.readthedocs.io/en/latest/developer/architecture.html#supported-execution-frameworks-and-memory-formats

File: docs/UsingOmnisci/index.rst (hunk 1 above)
Comment (Garra1980) [reply docs/UsingOmnisci/index.rst:1]: No, we don't use Ray de facto so let's not mention it

File: docs/index.rst (hunk 4 above)
Comment (Garra1980) [reply docs/index.rst:173]: Thanks! Done

File: docs/UsingOmnisci/index.rst (hunk 1 above)
Comment (YarShev) [reply docs/UsingOmnisci/index.rst:1]: Ok but let's plan refactor for omnisci backend for one of the near release to get rid of Ray if we don't want to use it. Currently, it looks confusing.

File: docs/UsingOmnisci/index.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+This section describes usage of related documents for the OmniSciDB-based engine of Modin.
+
+This engine uses analytical database OmniSciDB_ to obtain high single-node scalability for
+specific set of dataframe operations.
+To enable this engine you could set the following environment variables:
+
+.. code-block:: bash
+
+   export MODIN_ENGINE=ray
+   export MODIN_BACKEND=OmniSci
```
Comment (YarShev) [inline docs/UsingOmnisci/index.rst:13]:
```suggestion
   export MODIN_BACKEND=omnisci
```

let's use lower-casing for simplicity.

File: docs/UsingOmnisci/index.rst [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+or turn it on in source code:
+
+.. code-block:: python
+
+   import modin.experimental.pandas as pd
+   import modin.config as cfg
+   cfg.Engine.put('Ray')
```
Comment (YarShev) [inline docs/UsingOmnisci/index.rst:21]:
```suggestion
   cfg.Engine.put('ray')
```

File: docs/UsingOmnisci/index.rst [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+   cfg.Backend.put('OmniSci')
```
Comment (YarShev) [inline docs/UsingOmnisci/index.rst:22]:
```suggestion
   cfg.Backend.put('omnisci')
```

File: docs/developer/architecture.rst [hunk 8]
```diff
@@ -262,6 +262,9 @@ documentation page on :doc:`contributing </contributing>`.
 - :doc:`Pandas on Dask </UsingPandasonDask/index>`
     - Uses the `Dask Futures`_ execution framework.
     - The compute kernel/in-memory format is a pandas DataFrame.
+- :doc:`Omnisci </UsingOmnisci/index>`
+    - Uses OmniSciDB as an engine.
+    - The compute kernel/in-memory format is a pandas DataFrame.
```
Comment (YarShev) [inline docs/developer/architecture.rst:267]:
```suggestion
    - The compute kernel/in-memory format is a pyarrow Table or pandas DataFrame when defaulting to pandas.
```

cc @ienkovich

**Review by YarShev — APPROVED** [review]
LGTM

**Pull request merged**